admin/resources/js/components/classification/CreateClassification.tsx

```diff
@@ -0,0 +1,121 @@
+import { upperCase } from "lodash";
+import * as React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import {
+  CreateClassificationInput,
+  useCreateClassificationMutation,
+} from "../../api/generated";
+import errorMessages from "../form/errorMessages";
+import Input from "../form/Input";
+import Submit from "../form/Submit";
+
+type FormValues = CreateClassificationInput;
+interface CreateClassificationFormProps {
+  handleCreateClassification: (data: FormValues) => Promise<FormValues>;
+}
+
+export const CreateClassificationForm: React.FunctionComponent<CreateClassificationFormProps> =
+  ({ handleCreateClassification }) => {
+    const methods = useForm<FormValues>();
+    const { handleSubmit, watch } = methods;
+    const watchMinSalary = watch("minSalary");
+
+    const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+      const classification: FormValues = {
+        ...data,
+        group: upperCase(data.group),
+        level: Number(data.level),
+        minSalary: Number(data.minSalary),
+        maxSalary: Number(data.maxSalary),
+      };
+      return handleCreateClassification(classification)
+        .then(() => {
+          // TODO: Navigate to cmo asset dashboard
+        })
+        .catch(() => {
+          // Something went wrong with handleCreateClassification.
+          // Do nothing.
+        });
+    };
+    return (
+      <section>
+        <h2>Create Classification</h2>
+        <FormProvider {...methods}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <Input
+              id="name_en"
+              name="name.en"
+              label="Name: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="name_fr"
+              name="name.fr"
+              label="Name FR: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="group"
+              name="group"
+              label="Group: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="level"
+              name="level"
+              label="Level: "
+              type="number"
+              rules={{ required: errorMessages.required }}
```

Comment:
This allows for passing negative values or passing a long value that would return an error (for example: _GraphQL large integer error: Int cannot represent non 32-bit signed integer value_). I've never seen a level for any classification above 8, and so here are two possible solutions with that in mind:
- Add a rule where the value must be between 1 and 10
- Change the field from an `input` to a `select` and only allow options for 1 to 10

Reply:
That's a good point! I'll change it to a select. Now that I'm thinking about it that's what we used on TC as well.